addons/mail/models/discuss/ir_websocket.py

```diff
@@ -1,40 +1,46 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 import re
+from datetime import datetime, timedelta
 
 from odoo import models
+from odoo.tools import OrderedSet
+from odoo.osv import expression
 from odoo.addons.mail.models.discuss.mail_guest import add_guest_to_context
-from odoo.addons.mail.tools.discuss import Store
 
 
 class IrWebsocket(models.AbstractModel):
     _inherit = ["ir.websocket"]
 
-    def _get_missed_presences_identity_domains(self, presence_channels):
-        identity_domain = super()._get_missed_presences_identity_domains(presence_channels)
-        if guest_ids := [
-            g.id for g, _ in presence_channels if isinstance(g, self.pool["mail.guest"])
-        ]:
-            identity_domain.append([("guest_id", "in", guest_ids)])
-        return identity_domain
-
     def _get_missed_presences_bus_target(self):
         if self.env.user and not self.env.user._is_public():
-            return super()._get_missed_presences_bus_target()
+            return self.env.user.partner_id
         if guest := self.env["mail.guest"]._get_guest_from_context():
             return guest
         return None
 
     @add_guest_to_context
     def _build_presence_channel_list(self, presences):
-        channels = super()._build_presence_channel_list(presences)
+        """
+        Return the list of presences to subscribe to.
+
+        :param typing.List[typing.Tuple[str, int]] presences: The presence
```

Comment:
This parameter was complex because of the override. I didn't want to parse the channel list twice hence the list of tuple. I think we should remove this complexity and just pass the `channels` parameter of the `_build_bus_channel_list` method.